cnf-certification-test/networking/declaredandlistening_test.go

```diff
@@ -0,0 +1,328 @@
+			listeningPorts: map[declaredandlistening.Key]*provider.Container{},
+			container: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+			expectedlisteningPorts: map[declaredandlistening.Key]*provider.Container{},
+			expectedRes:            "",
+			expectContainer: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+		},
+		{
+			inputRes:       "\n",
+			listeningPorts: map[declaredandlistening.Key]*provider.Container{},
+			container: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+			expectedlisteningPorts: map[declaredandlistening.Key]*provider.Container{},
+			expectedRes:            "\n",
+			expectContainer: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+		},
+
+		{
+			inputRes:       "tcp LISTEN 0      128    0.0.0.0:8080 0.0.0.0:*\ntcp LISTEN 0      128    0.0.0.0:7878 0.0.0.0:*\n",
+			listeningPorts: map[declaredandlistening.Key]*provider.Container{},
+			container: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+			expectedlisteningPorts: map[declaredandlistening.Key]*provider.Container{{Port: 8080, Protocol: "TCP"}: {
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+				{Port: 7878, Protocol: "TCP"}: {
+					Data: &v1.Container{
+						Name: "",
+					},
+					Namespace: "",
+					Podname:   "",
+				},
+			},
+			expectedRes: "tcp LISTEN 0      128    0.0.0.0:8080 0.0.0.0:*\ntcp LISTEN 0      128    0.0.0.0:7878 0.0.0.0:*\n",
+			expectContainer: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+		},
+
+		{
+			inputRes:       "udp LISTEN 0      128    0.0.0.0:8080 0.0.0.0:*\nudp LISTEN 0      128    0.0.0.0:7878 0.0.0.0:*\n",
+			listeningPorts: map[declaredandlistening.Key]*provider.Container{},
+			container: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+			expectedlisteningPorts: map[declaredandlistening.Key]*provider.Container{{Port: 8080, Protocol: "UDP"}: {
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+				{Port: 7878, Protocol: "UDP"}: {
+					Data: &v1.Container{
+						Name: "",
+					},
+					Namespace: "",
+					Podname:   "",
+				},
+			},
+			expectedRes: "udp LISTEN 0      128    0.0.0.0:8080 0.0.0.0:*\nudp LISTEN 0      128    0.0.0.0:7878 0.0.0.0:*\n",
+			expectContainer: &provider.Container{
+				Data: &v1.Container{
+					Name: "",
+				},
+				Namespace: "",
+				Podname:   "",
+			},
+		},
+	}
+	for _, tc := range testCases {
+		declaredandlistening.ParseListening(tc.inputRes, tc.listeningPorts, tc.container)
```

Comment:
You can create a local variable here to store the listening ports that the function returns:
```
for _, tc := range testCases {
  listeningPorts := map[declaredandlistening.Key]*provider.Container
  declaredandlistening.ParseListening(tc.inputRes, listeningPorts, tc.container)
  assert.Equal(t, tc.expectedlisteningPorts, listeningPorts)
}
```

Reply:
I dont think need local variable.